src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableSingleOpEntity.java

```diff
@@ -176,6 +226,91 @@ public static ObTableSingleOpEntity getInstance(String[] rowKeyNames, Object[] r
         return entity;
     }
 
+    // Support class, which is used for column name sorted
+    private static class ColumnNamePair implements Comparable<ColumnNamePair> {
+        long number;
+        // we could use idx here, and adjust obj after compare
+        ObObj obj;
+
+        ColumnNamePair(long number, ObObj obj) {
+            this.number = number;
+            this.obj = obj;
+        }
+
+        @Override
+        public int compareTo(ColumnNamePair other) {
+            return Long.compare(this.number, other.number);
+        }
+    }
+
+    /*
+     * adjustRowkeyColumnName should be execute in the last
+     */
+    public void adjustRowkeyColumnName(Map<String, Long> columnNameIdxMap) {
+        this.rowKeyBitLen = columnNameIdxMap.size();
+        int size = (int) Math.ceil(columnNameIdxMap.size() / 8.0);
+        byte[] byteArray = new byte[size];
+        List<Long> columnNameIdx = new LinkedList<>();
+
+
+        for (String name : rowKeyNames) {
+            Long index = columnNameIdxMap.get(name);
+            columnNameIdx.add(index);
+            if (index != null) {
+                int byteIndex = index.intValue() / 8;
+                int bitIndex = index.intValue() % 8;
+                byteArray[byteIndex] |= (byte) (1 << bitIndex);
+            }
+        }
+
+        List<ColumnNamePair> pairs = new ArrayList<>();
+        for (int i = 0; i < columnNameIdx.size(); i++) {
+            pairs.add(new ColumnNamePair(columnNameIdx.get(i), rowkey.get(i)));
+        }
+
+        Collections.sort(pairs);
+
+        rowkey = new ArrayList<>(pairs.size());
+        for (ColumnNamePair pair : pairs) {
+            rowkey.add(pair.obj);
+        }
+
+        this.rowKeyBitMap = byteArray;
+    }
+
+    public void adjustPropertiesColumnName(Map<String, Long> columnNameIdxMap) {
+        if (!ignoreEncodePropertiesColumnNames) {
+            this.propertiesBitLen = columnNameIdxMap.size();
+        }
+        int size = (int) Math.ceil(columnNameIdxMap.size() / 8.0);
+        byte[] byteArray = new byte[size];
```

Comment:
this two functions can abstract into one function

Reply:
Fixed, set into `isSamePropertiesColumnNamesLen()`